readthedocs/core/resolver.py

```diff
@@ -146,36 +145,34 @@ def resolve_path(
 
         filename = self._fix_filename(filename)
 
-        parent_project, subproject_relationship = self._get_canonical_project_data(
-            project
-        )
+        parent_project, project_relationship = self._get_canonical_project_data(project)
         cname = (
             cname
             or self._use_subdomain()
             or parent_project.get_canonical_custom_domain()
         )
         single_version = bool(project.single_version or single_version)
 
-        # If the project is a subproject, we use the urlpattern
-        # of the child of the relationship. For a project that
-        # isn't a subproject, we use the urlpattern of the
-        # parent project.
-        if subproject_relationship:
-            urlpattern = subproject_relationship.child.urlpattern
+        # If the project is a subproject, we use the custom prefix
+        # of the child of the relationship, this is since the project
+        # could be a translation. For a project that isn't a subproject,
+        # we use the custom prefix of the parent project.
+        if project_relationship:
+            custom_prefix = project_relationship.child.custom_prefix
         else:
-            urlpattern = parent_project.urlpattern
+            custom_prefix = parent_project.custom_prefix
```

Comment:
I feel like this could cause some odd issues with subprojects having different URL patterns than the parent? I don't fully understand the thinking here around translations?

Reply:
All relationships have a parent project, for subprojects it's the main project, and for translations it's the main language project.

This is the same logic we use to "unresolve", so they both match (the unresolver first "unresolves" the parent project, and then it tries to resolve the relationship from that).